dice_ml/diverse_counterfactuals.py

```diff
@@ -82,7 +82,11 @@ def visualize_as_dataframe(self, display_sparse_df=True, show_only_changes=False
         print('Query instance (original outcome : %i)' %round(self.test_pred))
         display(self.org_instance) #  works only in Jupyter notebook
 
-        if 'data_df' in self.data_interface.__dict__ and display_sparse_df==True and self.final_cfs_sparse is not None:
+        if self.posthoc_sparsity_param == 'NA':
+            print('\Feasible Counterfactual set (new outcome : %i)' %(1-round(self.test_pred)))
```

Comment:
Okay I understand. you are using NA parameter to show feasible CFs. But then it's an odd choice: why should setting the sparsity to NA make the counterfactuals feasible?
One option is to add a top_level argument for feasibility. With that argument, you can ask the user for different kinds of inputs:
1. box constraints: to say that age lies between 0 to 100. 
2. other feasibility constraints

We will need to think about what is a consistent API for feasibility. One option is to create an optional parameter called feasibility constraints where a user can provide these constraints.